tools/cndump/fatal.c

```diff
@@ -0,0 +1,37 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/*
+ * Copyright (C) 2022 Micron Technology, Inc.  All rights reserved.
+ */
+
+#include <stdio.h>
+
+#include <hse/hse.h>
+
+#include "fatal.h"
+#include "globals.h"
+
+void
+fatal(const char *who, merr_t err)
+{
+    char buf[256];
+
+    hse_strerror(err, buf, sizeof(buf));
+    fprintf(stderr, "%s: %s: %s\n", progname, who, buf);
+    exit(1);
+}
+
+__attribute__((format(printf, 1, 2))) void
+syntax(const char *fmt, ...)
```

Comment:
Much prefer `syntax()` since it keeps the error message brief and to the point, instructs user to "use -h for help" and exits with the correct code (well, the next push will exit with EX_USAGE).